### DISCUSSION TITLE
How can put class into the store

### DISCUSSION THREAD

**Discussion by njfamirm** [alpinejs/alpine#4281] [1. Help] [answered]
Hi

currently i'm pass a object contain methods and some keys to the store as config and this is work correctly, but because the object not have strong type support and want to set type manually, i want to use class and pass class into the store, how can do this?

thanls

**Comment by njfamirm** [comment]
I'm use a solutions that's a class create a interall store and put some value in the this.store and pass this.store to the alpine.store(name, config), but when call some functions that's change this.store.x, the store not updated correctly in the ui!

**Comment by ekwoka** [comment]
Class instances will work just fine.

If you're having issues with it, we'd need to see some code to see what you're concerned about.

**Reply by njfamirm** [reply, marked as answer]
I'm found my issue again,

my mistake is I'm use a object that's bind to the store so this is not `Proxy`
When I'm use the returned object when call alpine.store(name), it's work correctly, thanks
